streaming/jms-connector/src/main/java/JmsMessageProducer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.hazelcast.jet.impl.util.ExceptionUtil;
+import org.apache.activemq.ActiveMQConnectionFactory;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import static com.hazelcast.jet.impl.util.Util.uncheckRun;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Utility class to produce messages to the given destination
+ */
+public class JmsMessageProducer {
+
+    private final Thread producerThread;
+
+    JmsMessageProducer(String destinationName, DestinationType destinationType) {
+        producerThread = new Thread(() -> {
+            ConnectionFactory connectionFactory = new ActiveMQConnectionFactory(ActiveMQBroker.BROKER_URL);
+            Connection connection = null;
+            try {
+                connection = connectionFactory.createConnection();
```

Comment:
we're not able to use Autocloseable here?

Reply:
`Connection` is not `Autocloseable`, ActiveMQ uses JMS API 1.1